File: Data Structures & Algorithms/same-binary-tree/submission-1.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public bool IsSameTree(TreeNode p, TreeNode q) {

        return dfs(p,q);
    
    }

    public bool dfs(TreeNode node1, TreeNode node2){
        if(node1 is null && node2 is null) return true;
        if(node1 is null && node2 != null) return false; 
        if(node2 is null && node1 != null) return false;

        
        var val1 = node1.val; 
        var val2 = node2.val;

        if(val1 != val2) return false;

        return dfs(node1.left,node2.left) && dfs(node1.right,node2.right);
    }
}

File: Data Structures & Algorithms/binary-tree-from-preorder-and-inorder-traversal/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def buildTree(self, preorder: List[int], inorder: List[int]) -> Optional[TreeNode]:
        if not preorder or not inorder:
            return None
        #get the root from preorder 

        root = TreeNode(preorder[0])

        node = root

        #mid of tree is the index of root of inOrder
        mid = inorder.index(root.val)

        node.left = self.buildTree(preorder[1 : mid + 1], inorder[:mid])
        node.right = self.buildTree(preorder[mid+1:], inorder[mid + 1:])
        
        

        return root
    
    #PreOrder - root, left , right 
    #InOrder - left, root, right
    #Tree has unique values